examples/network/common.c

```diff
@@ -20,15 +22,27 @@ int cred_acquire_cb(git_cred **out,
 		unsigned int UNUSED(allowed_types),
 		void * UNUSED(payload))
 {
-	char username[128] = {0};
-	char password[128] = {0};
+	char *username = NULL, *password = NULL;
+	int error;
 
 	printf("Username: ");
-	scanf("%s", username);
+	if (getline(&username, NULL, stdin) < 0) {
```

Comment:
According to my manpage, "This buffer should be freed by the user program even if `getline()` failed", not that it's a big deal in this case.